app/helpers/users_helper.rb

```diff
@@ -10,4 +10,31 @@ def gravatar_for(user, options = { size: 80 })
   def make_gravatar_id(user)
     Digest::MD5.hexdigest(user.email.downcase)
   end
+
```

Comment:
- [ ] There should be no application logic on the helpers. -- helper files are for view-related logic but not application-related logic.
Application logic usually hit the database to grab information like your methods here (friends, pending_friends, friends_request, ...) and it's place is not in the view helper. Views and View helpers are only responsible for displaying information that became available to them through controllers, they supposed to have no knowledge on how to get these data. As the name `View` suggests, it's for just **view**ing.

So these methods makes better sense to be as instance methods in `user` model.